internal/buffer/buffer.go

```diff
@@ -40,44 +40,22 @@ type Buffer struct {
 	next *Buffer
 }
 
-// Buffers manages the reuse of individual buffer instances. It is thread-safe.
-type Buffers struct {
-	// mu protects the free list. It is separate from the main mutex
-	// so buffers can be grabbed and printed to without holding the main lock,
-	// for better parallelization.
-	mu sync.Mutex
-
-	// freeList is a list of byte buffers, maintained under mu.
-	freeList *Buffer
+var buffers = sync.Pool{
```

Comment:
Then comparing performance using benchstat please ensure that `p` factor is lower than <0.05, so the results are statistically significant. Fact that benchstat returns ` ~` for time delta doesn't mean that results are equal. It means that there were not enough tries to make any conclusions. 

I recommend increase number of tries to when benchmarking to at last 10.

Reply:
That didn't help. For example:
```console
$ go test -bench=KlogOutput/verbosity_disabled -benchmem -count=10 . >/tmp/run1.log
$ go test -bench=KlogOutput/verbosity_disabled -benchmem -count=10 . >/tmp/run2.log
$ $GOPATH/bin/benchstat /tmp/run1.log /tmp/run2.log 
name                              old time/op    new time/op    delta
KlogOutput/verbosity_disabled-36    14.8ns ± 0%    14.8ns ± 0%   ~     (p=1.000 n=10+10)

name                              old alloc/op   new alloc/op   delta
KlogOutput/verbosity_disabled-36     0.00B          0.00B        ~     (all equal)

name                              old allocs/op  new allocs/op  delta
KlogOutput/verbosity_disabled-36      0.00           0.00        ~     (all equal)
```

There's no code change at all between the two invocations and the `14.8ns` duration is the same with very little variance (0%), but p remains at 1.000.

Reply:
High p value means that there is no statistically significant difference between both solutions. It makes sense as both options take exactly `14.8ns`. https://pkg.go.dev/golang.org/x/perf/cmd/benchstat#hdr-Tips

This is exactly what we would result to be if you would want to prove that solutions have same performance.

Interesting thing is that variance is so low. I would always expect at 1% variation.  Not sure why, maybe problem with benchmark?

Reply:
Now I am confused. Earlier you said:
> Fact that benchstat returns  ~ for time delta doesn't mean that results are equal. It means that there were not enough tries to make any conclusions.

But in this example it means exactly that: results are equal.

Reply:
Sorry for confusion, `~` just means it could not determine whether there is a statistically significant improvement difference between two runs. This happens when `p` is above 0.05 . This can mean either:
* There is a change, but we didn't get enough runs to be sure. For example we got `p` equal 0.5, which means we are 50% sure that there is an improvement.
* Results are equal. 

In first case increasing number of runs should result in `p` decreasing. In that case we should increase the number of runs until we get `p` below 0.05. 

As in example below. We can see that new time is on average 0.06µs faster. However we also see that this difference is around the about the result variance `± 6%`. So if we take best results from old time, they are still better then new time. In such situation we need to use statistical analysis to confirm the result. Thankfully benchstat already gives us the `p` value equal 0.246 which means that there is 75.4% (1-0.246) chance that there is an improvement. This however is usually not enough and we want to have at least 95% certainty (p value below 0.05). In this case we should increasing number of runs should help
```
name              old time/op    new time/op    delta
Header-36           1.68µs ± 7%    1.62µs ± 6%   ~     (p=0.246 n=5+5)
```



In second case p will never decrease no matter how many runs we do. Example:
```
name old time/op new time/op delta
KlogOutput/verbosity_disabled-36 14.8ns ± 0% 14.8ns ± 0% ~ (p=1.000 n=10+10)
```

Reply:
In second case results are suspicious. There are many factors that should cause execution time to vary, like cpu temperature, boost, background processes etc. However the point stand that increasing number of runs doesn't help. I would recommend double checking if benchmark is correctly implemented.

Reply:
> There are many factors that should cause execution time to vary, like cpu temperature, boost, background processes etc. 

I am running this on a high-quality desktop with nothing besides the benchmark running. I think it is valid that the results are stable.